lifting-state-up/single-source-of-truth.md

```diff
@@ -0,0 +1,260 @@
+## Single Source of Truth
+
+In software, we're reading, moving, and manipulating data all the time. ["Single source of truth"](https://en.wikipedia.org/wiki/Single_source_of_truth) is the practice of structuring our code so that data is managed in one place. All other places that use this data should use the data as read-only.
+
+In Sofia's app, all components that read student data, should reference it from the same source.
+
+Right now, Sofia's app introduces student data in two places:
+
+- The `App` component, which holds student names and emails
+- The `Student` component, which manages student `isPresent` details
+
+### !callout-info
+
+## The Flip Side: Managing Multiple Sources of "Truth"
+
+Consider for a moment what it would be like if we _didn't_ strive for a single source of truth. As we continued to develop our app, we'd possibly ask the following questions:
+
+- If another component needs to read a student's `isPresent` state, where do they look for that data?
+  - For example, what if another component wanted to count how many students were absent that day?
+- If there are multiple sources of truth, in what situations do we update one of source of truth, and don't update the others?
+- What happens if our multiple sources of truth are out of sync with each other?
+
+<br/>
+
+These questions highlight problems that are challenging to work with!
+
+### !end-callout
+
+## Moving `isPresent` From `Student` to `App`
+
+Sofia's first goal is to create a single source of truth for her app. She wants to move all student data to one place.
+
+Sofia chooses to move the data to the `App` component. She chooses the `App` component because:
+
+1. It's more appropriate for the responsibility we gave `App`
+1. In React, we pass read-only data through `props`, which is passed into nested components. `Student` is more nested than `App`.
+
+### New Student Data Shape in `App`
+
+First, we should add `isPresent` into our student data in `App`.
+
+While we're here, we're also going to add `id` data to each student. This helps the data feel a little more realistic, as if we're reading it from a database or API.
```

Comment:
What happened to Sofia? Bringing her back into the picture (throughout the lesson)
```suggestion
First, Sofia adds `isPresent` to the student data in `App`.

While she's there, she also adds `id` data to each student. This helps the data feel a little more realistic, as if it were coming from a database or API.
```